waku/protocol/v2/waku_filter.nim

```diff
@@ -140,6 +149,11 @@ proc subscription*(proto: WakuFilter): MessageNotificationSubscription =
 
   MessageNotificationSubscription.init(@[], handle)
 
-proc subscribe*(wf: WakuFilter, peer: PeerInfo, request: FilterRequest) {.async, gcsafe.} =
-  let conn = await wf.switch.dial(peer.peerId, peer.addrs, WakuFilterCodec)
-  await conn.writeLP(FilterRPC(requestId: generateRequestId(wf.rng), request: request).encode().buffer)
+proc subscribe*(wf: WakuFilter, request: FilterRequest): Future[string] {.async, gcsafe.} =
+  let id = generateRequestId(wf.rng)
+  if wf.peers.len >= 1:
+    let peer = wf.peers[0].peerInfo
+    # @TODO: THERE SHOULD BE ERROR HANDLING HERE, WHAT IF A PEER IS GONE? WHAT IF THERE IS A TIMEOUT ETC.
+    let conn = await wf.switch.dial(peer.peerId, peer.addrs, WakuFilterCodec)
```

Comment:
I think dialing would better be done somewhere else like in `init` or even outside of the protocol (So that a node can control better the amount of peers), but this can be adapted later.